pkg/minishift/registration/redhat.go

```diff
@@ -51,17 +52,36 @@ func (registrator *RedHatRegistrator) CompatibleWithDistribution(osReleaseInfo *
 }
 
 func (registrator *RedHatRegistrator) Register(param *RegistrationParameters) error {
-	if output, err := registrator.SSHCommand("sudo -E subscription-manager version"); err != nil {
+	output, err := registrator.SSHCommand("sudo -E subscription-manager version")
+	if err != nil {
 		return err
-	} else {
-		if strings.Contains(output, "not registered") {
+	}
+	if strings.Contains(output, "not registered") {
+		for i := 1; i < 4; i++ {
+			if param.Username == "" {
+				param.Username = param.GetUsernameInteractive("Username to register to RHSM")
```

Comment:
I see, is RHSM the official acronym for the subscription manager? I was wondering whether it should say something like RedHat Developers credentials, but that is not really required, right? Other subscriptions would work as well. I guess the confusion for people coming from the Red Hat Developers has always been that they sign up with Red Hat Developers and get under the hood a Red Hat subscription. For them this is not transparent though and for them RHSM might not make much sense.